kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/uploadable/PodUploadWebSocketListener.java

```diff
@@ -45,39 +43,28 @@
   }
 
   @Override
-  public void onOpen(WebSocket webSocket, Response response) {
+  public void onOpen(WebSocket webSocket) {
     webSocketRef.set(webSocket);
   }
 
   @Override
-  public void onMessage(WebSocket webSocket, ByteString bytes) {
-    if (readyLatch.getCount() > 0 && bytes.size() == 1) {
+  public void onMessage(WebSocket webSocket, ByteBuffer bytes) {
+    if (readyLatch.getCount() > 0 && bytes.remaining() == 1) {
       readyLatch.countDown();
-    } else if (bytes.size() > 1) {
-      error.set(bytes.substring(1).toString());
+    } else if (bytes.remaining() > 1) {
+      bytes.position(1);
+      error.set(StandardCharsets.UTF_8.decode(bytes).toString());
     }
   }
-
+  
   @Override
-  public void onClosed(WebSocket webSocket, int code, String reason) {
+  public void onClose(WebSocket webSocket, int code, String reason) {
     completeLatch.countDown();
   }
 
   @Override
-  public void onFailure(WebSocket webSocket, Throwable t, Response response) {
-    if (response != null) {
-      final String responseBody = Optional.ofNullable(response.body())
-        .map(rb -> {
-          try {
-            return rb.string();
-          } catch (IOException e) {
-            return e.getMessage();
-          }
-        }).orElse("");
-      error.set(String.format("%s - %s%n%s", response.code(), response.message(), responseBody));
-    } else {
-      error.set("PodUploadWebSocketListener failed with no response");
-    }
+  public void onError​(WebSocket webSocket, Throwable t) {
```

Comment:
There's another Zero Width Character between onError and (, maybe it's worth doing a quick find to remove them.

Reply:
```
Targets
    Occurrences of '​' in Project
Found Occurrences in Project  (13 usages found)
    ExecWebSocketListener.java  (2 usages found)
        188 onError​(webSocket, e);
        197 public void onError​(WebSocket webSocket, Throwable t) {
    OkHttpWebSocketImpl.java  (1 usage found)
        70 listener.onError​(new OkHttpWebSocketImpl(webSocket), t);
    PodOperationsImpl.java  (1 usage found)
        319 execWebSocketListener.onError​(w, t);
    PodUpload.java  (1 usage found)
        177 podUploadWebSocketListener.onError​(w, t);
    PodUploadWebSocketListener.java  (1 usage found)
        66 public void onError​(WebSocket webSocket, Throwable t) {
    PodUploadWebSocketListenerTest.java  (1 usage found)
        68 podUploadWebSocketListener.onError​(null, new IOException("here"));
    PortForwarderWebsocket.java  (2 usages found)
        274 public void onError​(WebSocket webSocket, Throwable t) {
        321 listener.onError​(w, t);
    README.md  (1 usage found)
        104 - kondoumh: [​Kubernetes ネイティブなカオスエンジニアリングツール Chaos Mesh を使ってみる](https://blog.kondoumh.com/entry/2020/10/23/123431)
    WatchConnectionManager.java  (1 usage found)
        125 listener.onError​(w, t);
    WatcherWebSocketListener.java  (1 usage found)
        49 public void onError​(WebSocket webSocket, Throwable t) {
    WebSocket.java  (1 usage found)
        35 default void onError​(WebSocket webSocket, Throwable error) {}
```

Reply:
I have no idea how that got in there.  Should be removed now.